class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>(); // value --> index 

        for (int i = 0; i < nums.length; i++)
        {
            map.put(nums[i], i);

        }
        System.out.println(map);

        for (int i = 0; i < nums.length; i++)
        {
            int diff = target - nums[i];
            if (map.containsKey(diff) && map.get(diff) != i)
            {
                return new int[]{i, map.get(diff)};
            }
        }

        return null;
    }
}
